onnxruntime/test/providers/cpu/math/quantize_linear_matmul_test.cc

```diff
@@ -21,42 +21,50 @@ TEST(QuantizeLinearMatmulOpTest, QLinearMatMul3D) {
 
   test.AddInput<uint8_t>("T2", {2, 4, 3},
                          {152, 51, 244,
-                          60, 26, 255, 
-                          0, 127, 246, 
+                          60, 26, 255,
+                          0, 127, 246,
                           127, 254, 247,
 
                           152, 51, 244,
                           60, 26, 255,
                           0, 127, 246,
                           127, 254, 247});
-    
+
   test.AddInput<float>("b_scale", {}, {0.00705f});
   test.AddInput<uint8_t>("b_zero_point", {}, {114});
 
   test.AddInput<float>("y_scale", {}, {0.0107f});
   test.AddInput<uint8_t>("y_zero_point", {}, {118});
-  test.AddOutput<uint8_t>("T3", {2, 2, 3}, 
+  test.AddOutput<uint8_t>("T3", {2, 2, 3},
                           {168, 115, 255,
                            1, 66, 151,
 
                            168, 115, 255,
                            1, 66, 151});
-                           
+
   test.Run();
 }
 
-TEST(QuantizeLinearMatmulOpTest, QLinearMatMul) {
+static void QLinearMatMul2DTest(bool only_t1_not_initializer) {
   OpTester test("QLinearMatMul", 10);
   test.AddInput<uint8_t>("T1", {2, 4}, {208, 236, 0, 238, 3, 214, 255, 29});
-  test.AddInput<float>("a_scale", {}, {0.0066f});
-  test.AddInput<uint8_t>("a_zero_point", {}, {113});
-  test.AddInput<uint8_t>("T2", {4, 3}, {152, 51, 244, 60, 26, 255, 0, 127, 246, 127, 254, 247});
-  test.AddInput<float>("b_scale", {}, {0.00705f});
-  test.AddInput<uint8_t>("b_zero_point", {}, {114});
-  test.AddInput<float>("y_scale", {}, {0.0107f});
-  test.AddInput<uint8_t>("y_zero_point", {}, {118});
+  test.AddInput<float>("a_scale", {1}, {0.0066f}, only_t1_not_initializer);
+  test.AddInput<uint8_t>("a_zero_point", {1}, {113}, only_t1_not_initializer);
+  test.AddInput<uint8_t>("T2", {4, 3}, {152, 51, 244, 60, 26, 255, 0, 127, 246, 127, 254, 247}, only_t1_not_initializer);
+  test.AddInput<float>("b_scale", {1}, {0.00705f}, only_t1_not_initializer);
+  test.AddInput<uint8_t>("b_zero_point", {1}, {114}, only_t1_not_initializer);
+  test.AddInput<float>("y_scale", {1}, {0.0107f}, only_t1_not_initializer);
+  test.AddInput<uint8_t>("y_zero_point", {1}, {118}, only_t1_not_initializer);
   test.AddOutput<uint8_t>("T3", {2, 3}, {168, 115, 255, 1, 66, 151});
   test.Run();
 }
+
+TEST(QuantizeLinearMatmulOpTest, QLinearMatMul) {
+  QLinearMatMul2DTest(false);
+}
+
+TEST(QuantizeLinearMatmulOpTest, QLinearMatMulAllInputExceptT1AreInitializers) {
+  QLinearMatMul2DTest(true);
```

Comment:
Would be good to add a comment noting that this is for NNAPI testing (like you did in quantize_ops_test.cc) to all the places you've added this test variant

Reply:
Added comments, but will keep these extra test cased running, the reason I disabled the one in  quantize_ops_test.cc for EP other than NNAPI is because that one will fail for TensorRT, (maybe a bug?)

---
In reply to: [456965962](https://github.com/microsoft/onnxruntime/pull/4530#discussion_r456965962) [](ancestors = 456965962)